heidrun/nypl_mods.rb

```diff
@@ -0,0 +1,285 @@
+    # here: http://www.loc.gov/marc/relators/relaterm.html
+    creator class: DPLA::MAP::Agent,
+            each: record.field('name')
+                    .select { |name|
+                      name['role']
+                                .field('roleTerm')
+                                .match_attribute(:type, 'text')
+                                .map { |v| v.value.downcase }
+                                .all? { |role_term|
+                                  CREATOR_ROLES.include?(role_term)
+                                }
+                    },
+            as: :contrib do
+      providedLabel contrib.field('namePart')
+    end
+
+    date record.map(&build_dates).flatten
+
+    description record.fields(['note'], %w(physicalDescription note))
+
+    extent record.field('physicalDescription', 'extent')
+
+    dcformat record.field('physicalDescription', 'form')
+
+    genre record.field('genre').select { |genre|
+      genre.value.downcase =~ /book|periodical|magazine/
+    }
+
+    # Top-level <identifier> when displayLabel OR type attribute contain the
+    #   following values: local_imageid isbn isrc isan ismn iswc issn uri urn
+    identifier record.field('identifier')
+                .select { |val|
+                  pattern = Regexp.union('local_imageid', 'isbn', 'isrc',
+                                         'isan', 'ismn', 'iswc', 'issn',
+                                         'uri', 'urn')
+                  val.node['displayLabel'] =~ pattern ||
+                    val.node['type'] =~ pattern
+                }
+
+    language class: DPLA::MAP::Controlled::Language,
+             each: record.field('language', 'languageTerm')
+                     .match_attribute(:type, 'code')
+                     .match_attribute(:authority, 'iso639-2b'),
+             as: :lang do
+      providedLabel lang
+    end
+
+    spatial class: DPLA::MAP::Place,
+            each: record.field('place', 'placeTerm'),
+            as: :place do
+      providedLabel place.field('geographic')
+    end
+
+    # TODO: Might need a deduping enrichment here, as providedLabel might be
+    # duplicated in some cases (as it is for
+    # http://dp.la/item/16ed9c7ba1efb5f5311cb188e09be881)
+    publisher class: DPLA::MAP::Agent,
+              each: record.field('originInfo', 'publisher'),
+              as: :origin_publisher do
+      providedLabel origin_publisher
+    end
+
+    relation record.map(&build_relation).flatten
+
+    rights record.field('rightsStatement')
+
+    subject class: DPLA::MAP::Concept,
+            each: record.fields(['subject'], ['genre']),
+            as: :subject do
+      providedLabel subject.map { |i| i.value.strip }
+    end
+
+    temporal class: DPLA::MAP::TimeSpan,
+             each: record.field('subject', 'temporal'),
+             as: :date_string do
+      providedLabel date_string
+    end
+
+    title record.map(&build_title).flatten
+
+    # TODO: Enrichment needed for DCMI
```

Comment:
Would this be different from the basic set of DCMI Type enrichments

https://github.com/dpla/KriKri/blob/develop/lib/krikri/enrichments/dcmi_type_map.rb 
https://github.com/dpla/KriKri/blob/develop/lib/krikri/enrichments/dcmi_enforcer.rb